### PR TITLE
feat(plugins): generic per-test scope hooks in `it` and `test`

### PR DESCRIPTION
**Pull request by Lojhan**
## Summary
This PR introduces a generic per-test scope lifecycle in Poku so plugin state can be isolated safely, even when tests run concurrently inside the same describe block.

Instead of relying on module-level globals that can leak between tests, Poku now exposes a neutral scope hook contract that any plugin can register and use. Core remains plugin-agnostic.

## Why we are building this
Today, tests can already execute concurrently in the same file when multiple it/test calls are registered synchronously.  
When plugin internals use shared process-level state (for example module-level Set/Map registries), one test can accidentally mutate or clean up resources owned by another test.

This creates:
- cross-test leakage
- flaky behavior
- cleanup races
- hard-to-debug failures under concurrency

We need a first-class per-test state boundary in the runner lifecycle, not only ad-hoc cleanup patterns in each plugin.

## What this feature adds
1. Generic scope hook contract in core test execution
- Added in src/modules/helpers/it/core.ts
- Core checks for a globally-registered scope provider and, if present, executes test callbacks through it.
- If no provider is registered, behavior is unchanged.

2. Neutral, plugin-agnostic integration point
- Hook key is generic and owned by Poku naming:
  - Symbol.for(@pokujs/poku.test-scope-hooks)

3. Concurrency and isolation verification
- Added unit test coverage in test/unit/scope-hooks.test.ts
- Confirms:
  - fallback path works with no hooks
  - concurrent runs receive independent stores
  - store identity remains stable across async boundaries

## Design goals achieved
- Core remains framework/plugin agnostic
- Backward compatible (no hooks => old behavior)
- Safe under concurrent test execution
- Extensible contract plugins can adopt incrementally

## What plugins can build from this
With this scope lifecycle, plugins can safely implement per-test resource ownership without global leaks, including:

- DOM framework adapters (mounted roots/fixtures per test)
- mock/spy registries with guaranteed scoped restore
- process/service lifecycle managers with scoped teardown
- DB transaction/session scopes per test
- request/tenant context propagation via AsyncLocalStorage
- scoped event/log capture and metrics buffers

In short: plugins can now implement reliable per-test state machines that are race-safe by default.

## Backward compatibility
- No breaking API changes for test authors
- Existing plugins that do not register scope hooks continue to run unchanged
- Hook registration is optional and additive

## Validation
- Focused unit test run for scope hooks passes:
  - test/unit/scope-hooks.test.ts

## Risks and mitigations
Risk:
- Incorrect plugin hook implementation could mis-handle cleanup order.

Mitigation:
- Core contract is minimal and explicit
- Test coverage verifies isolation behavior
- Fallback execution path remains intact when hooks are absent

## Follow-up work
- Adopt this generic scope contract in framework plugins (React, Vue, Angular) for mounted resource tracking
- Add helper utilities in shared packages to reduce plugin boilerplate when implementing scoped stores
- Expand integration tests for mixed plugin scenarios under concurrent executionContinued with AutopilotPrepared a complete, ready-to-paste PR description covering the feature purpose, motivation, architecture, plugin extensibility, compatibility, validation, risks, and follow-up roadmap.

### CODE REVIEW

File: src/modules/helpers/it/core.ts [hunk 1]
```diff
@@ -6,6 +6,17 @@ import { errorHoist, GLOBAL } from '../../../configs/poku.js';
...
+const SCOPE_HOOKS_KEY = Symbol.for('@pokujs/poku.test-scope-hooks');
+type ScopeHooks = {
+  createHolder: () => { scope: unknown };
+  runScoped: (
+    holder: { scope: unknown },
+    fn: () => Promise<unknown> | unknown
+  ) => Promise<void>;
+};
+const getScopeHooks = (): ScopeHooks | undefined =>
+  (globalThis as Record<symbol, ScopeHooks | undefined>)[SCOPE_HOOKS_KEY];
```
Comment (wellwelwel) [inline src/modules/helpers/it/core.ts]: How about:
```ts
const scopeHooks: ScopeHooks | undefined = (
  globalThis as Record<symbol, ScopeHooks | undefined>
)[SCOPE_HOOKS_KEY];
```
?

File: src/modules/helpers/it/core.ts [hunk 2]
```diff
@@ -57,8 +68,14 @@ export const itBase = async (
...
-      const resultCb = cb!();
-      if (resultCb instanceof Promise) await resultCb;
+      const hooks = getScopeHooks();
+      if (hooks) {
+        const holder = hooks.createHolder();
+        await hooks.runScoped(holder, () => cb!());
+      } else {
+        const resultCb = cb!();
+        if (resultCb instanceof Promise) await resultCb;
+      }
```
[the remark this answers is not in the document]
Comment (Lojhan) [reply src/modules/helpers/it/core.ts]: This is in general worse than something like:

```
let hasInitializedScopeHooks = false;
let cachedScopeHooks: ScopeHooks | undefined;

const getScopeHooks = (): ScopeHooks | undefined => {
  if (hasInitializedScopeHooks) return cachedScopeHooks;
...
```

as it generates a lookup on a generaly megamorphic variable that is globalThis. the usage of `hasInitializedScopeHooks` creates a housekeeping instance that yields the same result without any object property lookup overhead.

**Review by wellwelwel — DISMISSED** [review]

**Review by wellwelwel — APPROVED** [review]
